services/api/src/utils/middleware/error-handler.js

```diff
@@ -6,14 +6,17 @@ async function errorHandler(ctx, next) {
   try {
     await next();
   } catch (err) {
-    let { status, message, details } = err;
+    let { status = 500, message, details, expose } = err;
 
-    if (!status) {
+    // `expose` is set if the error is triggered via ctx.throw or ctx.assert
+    // if its not set => assume its 3 party error and should trigger a 500
```

Comment:
this might be nothing more than a matter of semantics but I don't think we can/should assume that it's only "third party" errors that will be thrown.. plenty of our own code can/will throw errors... of course maybe any code throwing outside a koa context could be considered "third party" so again it might just be semantics

Reply:
It's not a problem if we throw, only if you expect it not be 500.

e.g. throw Error("soemthing") will be 500;

naturally you can do the following if you really want 

```
const error = new Error("something")
error.expose = true;
error.status = 400;
throw error;
```

Let me know if that works for you ?

Reply:
I guess the only question is can a Koa thrown error with expose = false and a status < 500 make it through here? If so what will happen?

Reply:
I am thinking its possible if you do 

```
const error = new Error("something")
error.expose = false;
error.status = 400;
ctx.throw(err)
```

let me improve that check just in case